File: factors/python_factors/factors.py
import sys
import math

def factors_not_working(n):
  print ("1")
  #for idx_i in range(n - 1, 10):
  test = n - 1
  while(test >= math.sqrt(n)):
    print("number=%d" % test)
    if (n % test) == 0:
      print(int(n/test))
    test = test - 1
  print(n)

def factors(n):
  for idx_i in range(1, n + 1):
    if (n%idx_i) == 0:
      print(int(idx_i))


def main():
  if len(sys.argv) != 1:
    print ("Usage: %s <noargs>" % sys.argv[0])
    sys.exit()
  factors(100)
  sys.exit(), end

main()

File: factors/ocaml_factors/factors.ml
let rec range = function 0 -> [] | n -> range(n-1) @ [n]

let factors n =
  List.filter (fun v -> (n mod v) = 0) (range n)

let main () =
  if  Array.length Sys.argv <> 1 then
    begin
      Printf.printf "%s <noargs>\n" Sys.argv.(0);
      exit 1
    end
  else
    begin
      for idx_i = 1 to 10 do
        (*Printf.printf "calculateEasterSunday(%d)=<%s>\n" idx_i (calculateEasterSunday(idx_i)) *)
      done;
    end
let _ = main ()
